From 085e8b4e0e1268ab82245e3433fb33399720b7ff Mon Sep 17 00:00:00 2001
From: Lorenzo Colitti <lorenzo@colitti.com>
Date: Mon, 3 Jun 2024 23:25:25 +0900
Subject: [PATCH] configs: enable CONFIG_NET_FOU_IP_TUNNELS

The current configs enable fou as a module, but fou is not very
useful without CONFIG_NET_FOU_IP_TUNNELS, because it can't be
used to actually create tunnels:

$ sudo ip fou add port 12345 ipproto ipip
$ sudo ip link add name fou1 type ipip local any remote 8.8.8.8 \
      encap fou encap-sport auto encap-dport 12345
RTNETLINK answers: Invalid argument

Enable the option by default. This does not increase memory
footprint because fou is not loaded by default. Uncompressed
module size goes from 26.3KiB to 32.5KiB.

After this change, things work:

$ sudo ip fou add port 12345 ipproto ipip
$ sudo ip link add name fou1 type ipip local any remote 8.8.8.8 \
      encap fou encap-sport auto encap-dport 12345
$ ip link show fou1
8: fou1@NONE: <POINTOPOINT,NOARP> mtu 1472 qdisc noop state DOWN [...]
    link/ipip 0.0.0.0 peer 8.8.8.8

Signed-off-by: Lorenzo Colitti <lorenzo@colitti.com>
---
 arch/arm/configs/bcm2709_defconfig   | 2 +-
 arch/arm/configs/bcm2711_defconfig   | 2 +-
 arch/arm/configs/bcmrpi_defconfig    | 2 +-
 arch/arm64/configs/bcm2711_defconfig | 2 +-
 arch/arm64/configs/bcm2712_defconfig | 2 +-
 arch/arm64/configs/bcmrpi3_defconfig | 2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/configs/bcm2709_defconfig b/arch/arm/configs/bcm2709_defconfig
index f54eaec96297..08fecfbdfd47 100644
--- a/arch/arm/configs/bcm2709_defconfig
+++ b/arch/arm/configs/bcm2709_defconfig
@@ -95,7 +95,7 @@ CONFIG_IP_MROUTE_MULTIPLE_TABLES=y
 CONFIG_IP_PIMSM_V1=y
 CONFIG_IP_PIMSM_V2=y
 CONFIG_NET_IPVTI=m
-CONFIG_NET_FOU=m
+CONFIG_NET_FOU_IP_TUNNELS=y
 CONFIG_INET_AH=m
 CONFIG_INET_ESP=m
 CONFIG_INET_IPCOMP=m
diff --git a/arch/arm/configs/bcm2711_defconfig b/arch/arm/configs/bcm2711_defconfig
index c21271866954..26ca2e35ebe8 100644
--- a/arch/arm/configs/bcm2711_defconfig
+++ b/arch/arm/configs/bcm2711_defconfig
@@ -95,7 +95,7 @@ CONFIG_IP_MROUTE_MULTIPLE_TABLES=y
 CONFIG_IP_PIMSM_V1=y
 CONFIG_IP_PIMSM_V2=y
 CONFIG_NET_IPVTI=m
-CONFIG_NET_FOU=m
+CONFIG_NET_FOU_IP_TUNNELS=y
 CONFIG_INET_AH=m
 CONFIG_INET_ESP=m
 CONFIG_INET_IPCOMP=m
diff --git a/arch/arm/configs/bcmrpi_defconfig b/arch/arm/configs/bcmrpi_defconfig
index 4c2cf05be22b..0b72d3b9fd3b 100644
--- a/arch/arm/configs/bcmrpi_defconfig
+++ b/arch/arm/configs/bcmrpi_defconfig
@@ -90,7 +90,7 @@ CONFIG_IP_MROUTE_MULTIPLE_TABLES=y
 CONFIG_IP_PIMSM_V1=y
 CONFIG_IP_PIMSM_V2=y
 CONFIG_NET_IPVTI=m
-CONFIG_NET_FOU=m
+CONFIG_NET_FOU_IP_TUNNELS=y
 CONFIG_INET_AH=m
 CONFIG_INET_ESP=m
 CONFIG_INET_IPCOMP=m
diff --git a/arch/arm64/configs/bcm2711_defconfig b/arch/arm64/configs/bcm2711_defconfig
index f5e4a1263c6e..f225966db21d 100644
--- a/arch/arm64/configs/bcm2711_defconfig
+++ b/arch/arm64/configs/bcm2711_defconfig
@@ -103,7 +103,7 @@ CONFIG_IP_MROUTE_MULTIPLE_TABLES=y
 CONFIG_IP_PIMSM_V1=y
 CONFIG_IP_PIMSM_V2=y
 CONFIG_NET_IPVTI=m
-CONFIG_NET_FOU=m
+CONFIG_NET_FOU_IP_TUNNELS=y
 CONFIG_INET_AH=m
 CONFIG_INET_ESP=m
 CONFIG_INET_IPCOMP=m
diff --git a/arch/arm64/configs/bcm2712_defconfig b/arch/arm64/configs/bcm2712_defconfig
index df8f058a1ddf..bf1ea6d9da62 100644
--- a/arch/arm64/configs/bcm2712_defconfig
+++ b/arch/arm64/configs/bcm2712_defconfig
@@ -106,7 +106,7 @@ CONFIG_IP_MROUTE_MULTIPLE_TABLES=y
 CONFIG_IP_PIMSM_V1=y
 CONFIG_IP_PIMSM_V2=y
 CONFIG_NET_IPVTI=m
-CONFIG_NET_FOU=m
+CONFIG_NET_FOU_IP_TUNNELS=y
 CONFIG_INET_AH=m
 CONFIG_INET_ESP=m
 CONFIG_INET_IPCOMP=m
diff --git a/arch/arm64/configs/bcmrpi3_defconfig b/arch/arm64/configs/bcmrpi3_defconfig
index 38ee19e19f46..1193daa46f3b 100644
--- a/arch/arm64/configs/bcmrpi3_defconfig
+++ b/arch/arm64/configs/bcmrpi3_defconfig
@@ -97,7 +97,7 @@ CONFIG_IP_MROUTE=y
 CONFIG_IP_MROUTE_MULTIPLE_TABLES=y
 CONFIG_IP_PIMSM_V1=y
 CONFIG_IP_PIMSM_V2=y
-CONFIG_NET_FOU=m
+CONFIG_NET_FOU_IP_TUNNELS=y
 CONFIG_INET_AH=m
 CONFIG_INET_ESP=m
 CONFIG_INET_IPCOMP=m
-- 
2.34.1

